app/api/v2/managers/operation_api_manager.py

```diff
@@ -1,16 +1,45 @@
+import asyncio
+
+from marshmallow.schema import SchemaMeta
+from typing import Any
+
 from app.api.v2.managers.base_api_manager import BaseApiManager
-from app.api.v2.responses import JsonHttpNotFound, JsonHttpForbidden
+from app.api.v2.responses import JsonHttpNotFound, JsonHttpForbidden, JsonHttpBadRequest
+from app.objects.c_adversary import Adversary
+from app.objects.c_operation import Operation
+from app.utility.base_world import BaseWorld
 
 
 class OperationApiManager(BaseApiManager):
-    def __init__(self, data_svc, file_svc):
-        super().__init__(data_svc=data_svc, file_svc=file_svc)
+    def __init__(self, services):
+        super().__init__(data_svc=services['data_svc'], file_svc=services['file_svc'])
+        self.services = services
 
     async def get_operation_report(self, operation_id: str, access: dict):
         operation = await self.get_operation_object(operation_id, access)
         report = await operation.report(file_svc=self._file_svc, data_svc=self._data_svc)
         return report
 
+    async def create_object_from_schema(self, schema: SchemaMeta, data: dict,
+                                        access: BaseWorld.Access, existing_operation: Operation = None):
+        if data.get('state'):
+            await self.validate_operation_state(data, existing_operation)
+        obj_schema = schema()
+        operation = obj_schema.load(data)
```

Comment:
is there a way we can pre-process `data` to look up adversaries, planners, and other nested schema objects by ID and then load the appropriate data?

Reply:
Just updated this - the new `setup_operation()` function will use the unique attributes of planners/adversaries/agents to look up existing objects. If they don't exist, it will create default objects, similarly to how operations are currently created on the UI